### PR TITLE
Added two statistics for nop synthesis

### PR DESCRIPTION
**Pull request by zhengyangl**

### CODE REVIEW

File: lib/Pass/Pass.cpp [hunk 1]
```diff
@@ -28,12 +30,16 @@
 #include "llvm/Transforms/IPO/PassManagerBuilder.h"
 #include "llvm/Transforms/Utils/BasicBlockUtils.h"
 #include "llvm/Transforms/Utils/ModuleUtils.h"
+#include "llvm/ADT/Statistic.h"
```
Comment (chenyang78) [inline lib/Pass/Pass.cpp]: Maintain alphabetical order?

File: lib/Pass/Pass.cpp [hunk 2]
```diff
@@ -174,9 +180,13 @@ struct SouperPass : public ModulePass {
         if (V->getType() != T)
           continue;
         if (auto IP = dyn_cast<Instruction>(V)) {
-          // Domination check
-          if (DT.dominates(IP, ReplacedInst))
+          // Dominance check
+          if (DT.dominates(IP, ReplacedInst)) {
+            InstReplaced ++;
```
Comment (chenyang78) [inline lib/Pass/Pass.cpp]: Nit: redundant space before "++"

File: lib/Pass/Pass.cpp [hunk 3]
```diff
@@ -174,9 +180,13 @@ struct SouperPass : public ModulePass {
...
             return V;
+          } else {
+            InstReplaceFailed ++;
```
Comment (chenyang78) [inline lib/Pass/Pass.cpp]: Ditto

File: lib/Pass/Pass.cpp (hunk 1 above)
Comment (zhengyangl) [reply lib/Pass/Pass.cpp]: Thanks for the review!

File: lib/Pass/Pass.cpp (hunk 2 above)
Comment (zhengyangl) [reply lib/Pass/Pass.cpp]: Fixed.

File: lib/Pass/Pass.cpp (hunk 3 above)
Comment (zhengyangl) [reply lib/Pass/Pass.cpp]: Fixed.

**Comment by regehr** [comment]
Looks great-- please squash into a single commit

**Comment by zhengyangl** [comment]
Squashed!

**Pull request merged**